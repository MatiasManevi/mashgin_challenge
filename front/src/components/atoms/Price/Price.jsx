import React from 'react';
import clsx from 'clsx';

import './Price.scss';

const Price = ({ price, currency = 'u$s', small = false }) => {
	return (
		<div className="price">
			<p className={clsx({ small })}>
				{currency} {price}
			</p>
		</div>
	);
};

export default Price;
