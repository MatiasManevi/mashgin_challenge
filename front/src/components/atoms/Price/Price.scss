@import 'assets/scss/variables';

.price {
	p {
		font-size: $font-size-28;
		color: $color-white;
		&.small {
			font-size: $font-size-18;
		}
	}
}
