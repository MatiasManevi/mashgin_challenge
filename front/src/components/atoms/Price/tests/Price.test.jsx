import React from 'react';
import { shallow } from 'enzyme';

import Price from '../Price';

describe('Price', () => {
	const price = 2000;

	it('should match snapshot', () => {
		const wrapper = shallow(<Price price={price} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should use the right class', () => {
		const wrapper = shallow(<Price price={price} small />);
		expect(wrapper.find('p').props().className).toBe('small');
	});
});
