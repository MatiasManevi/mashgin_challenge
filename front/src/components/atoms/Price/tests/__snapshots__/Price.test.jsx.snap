// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price should match snapshot 1`] = `
<div
  className="price"
>
  <p
    className=""
  >
    u$s
     
    2000
  </p>
</div>
`;
